base/src/main/java/com/getbouncer/cardscan/base/MachineLearningThread.java

```diff
@@ -126,25 +178,31 @@ private synchronized RunArguments getNextImage() {
         return queue.pop();
     }
 
-    private void runModel() {
-        final RunArguments args = getNextImage();
-
-        Bitmap bm;
-        if (args.mFrameBytes != null) {
-            bm = getBitmap(args.mFrameBytes, args.mWidth, args.mHeight, args.mFormat,
-                    args.mSensorOrientation, args.mRoiCenterYRatio);
-        } else if (args.mBitmap != null) {
-            bm = args.mBitmap;
-        } else {
-            bm = Bitmap.createBitmap(480, 302, Bitmap.Config.ARGB_8888);
-            Canvas canvas = new Canvas(bm);
-            Paint paint = new Paint();
-            paint.setColor(Color.GRAY);
-            canvas.drawRect(0.0f, 0.0f, 480.0f, 302.0f, paint);
-        }
-
-        final Bitmap bitmap = bm;
+    private void runObjectModel(final Bitmap bitmap, final RunArguments args) {
+        // don't do anything for now
+        Handler handler = new Handler(Looper.getMainLooper());
+        handler.post(new Runnable() {
+            public void run() {
+                try {
+                    if (args.mObjectListener != null) {
+                        /*
+                        if (hadUnrecoverableException) {
+                            args.mScanListener.onFatalError();
+                        } else {
+                            args.mScanListener.onPrediction(number, ocr.expiry, bitmap, ocr.digitBoxes,
+                                    ocr.expiryBox);
+                        }*/
+                        args.mObjectListener.onPrediction(bitmap);
```

Comment:
We will need to add a few fields here.

Reply:
yeah, definitely, I just wasn't sure what these were :)